src/components/schedule-page/validation-drawer/validation-drawer.component.tsx

```diff
@@ -2,39 +2,82 @@ import Drawer from "../../common-components/drawer/drawer.component";
 import React, { useEffect, useState } from "react";
 import { useSelector } from "react-redux";
 import { ApplicationStateModel } from "../../../state/models/application-state.model";
-import {
-  ScheduleErrorLevel,
-  ScheduleErrorMessageModel,
-} from "../../../common-models/schedule-error-message.model";
-import { SpanErrors } from "./span-errors.component";
-import ErrorList from "./error-list.component";
+import { ScheduleErrorMessageModel } from "../../../common-models/schedule-error-message.model";
+import ErrorLoader from "./error-loader.component";
 
 export interface ValidationDrawerOptions {
   open: boolean;
   setOpen: (boolean) => void;
 }
 
+export enum errorLoaderState {
+  CHECKING = "CHECKING",
+  NOERRORS = "NOERRORS",
+  ERRORS = "ERRORS",
+  CONNECTIONFAILED = "CONNECTIONFAILED",
+}
+
+export interface CheckingStateProps {
+  state: errorLoaderState.CHECKING;
+  message: string;
+}
+
+export interface ErrorsStateProps {
+  state: errorLoaderState.ERRORS;
+  message: string;
+}
+
+export interface NoErrorsStateProps {
+  state: errorLoaderState.NOERRORS;
+  message: string;
+}
+
+export interface ConnectionFailedStateProps {
+  state: errorLoaderState.CONNECTIONFAILED;
+  message: string;
+}
```

Comment:
Myślę, że może lepiej by było użyć jeden interfejs który wygląda mniej więcej tak:
```
interface ValidationDrawerState {
    state: errorLoaderState;
    message: string;
} 
```
Czy myślisz, że te interfacy mogą bardziej się różnić i ma sens zostawić to tak jak jest teraz?

Reply:
To nie było jakoś specjalnie przemyślane: najpierw napisałam jeden, żeby zobaczyć czy działa, potem drugi itd. bo działało XD Wspólny interfejs jest zdecydowanie lepszym pomysłem :smile: